packages/core/docs/commercetools/migrate/1.4.0/index.md

```diff
@@ -0,0 +1,124 @@
+# Upgrading to 1.4.0
+
+## Introduction
+
+:::warning Start by updating the core packages
+This guide only includes commercetools-specific changes. Follow the [Migrating projects to 2.5.0](/migrate/2.5.0/overview.html) guide to update the core before you proceed with applying changes from this guide.
+:::
+
+In this release, we focused on improving security and performance and updating GraphQL types.
+
+This release includes some breaking changes. However, adjusting your projects shouldn't take too long and only requires a few minor changes.
+
+### Updated GraphQL types
+
+You might have noticed that the `@vue-storefront/commercetools-api` package exports [TypeScript types](/commercetools/api-reference/commercetools-api.html) for GraphQL data returned by commercetools. We didn't update them for quite a while and decided that it was time to do it. There are plenty of new types as well as some breaking changes to the old types. Depending on the types you use, you might need to update your project.
+
+### Handling access tokens
+
+We completely rewrote internals for handling authentication because of the poor performance of the old solution. Previously, the application requested a new access token several times for every user visiting the page if they didn't have the token cookie. This approach didn't make much sense, especially because [all guests share the same access token](https://docs.commercetools.com/api/authorization#create-anonymous-sessions-only-once-necessary) until their session become unique in any way, for example, by logging in or adding an item to the cart or wishlist.
+
+From now on, the application will request one access token for server operations (introduced in the previous release) and one for all guests. The server stores these tokens and doesn't send them out to the users. It will only create session cookies only for anonymous and logged-in users.
```

Comment:
```suggestion
From now on, the application will request one access token for server operations (introduced in the previous release) and one for all guests. The server stores these tokens and doesn't send them out to the users. It will create session cookies only for anonymous and logged-in users.
```